apollo-backend/src/main/java/io/logz/apollo/blockers/BlockerService.java

```diff
@@ -110,6 +125,38 @@ private boolean isUserAllowedToOverride(Deployment deployment, Blocker blocker)
 
     @SuppressWarnings("RedundantIfStatement")
     private boolean isBlockerInScope(Blocker blocker, Deployment deployment) {
+        Integer environmentToCheck = null;
+        Integer serviceToCheck = null;
+
+        if (blocker.getEnvironmentId() != null) {
+            environmentToCheck = blocker.getEnvironmentId();
+        }
+
+        if (blocker.getServiceId() != null) {
+            serviceToCheck = blocker.getServiceId();
+        }
+
+            if (blocker.getStackId() != null) {
+                switch (stackController.getStackType(blocker.getStackId())) {
+                    case ENVIRONMENTS:
+                        if (environmentsStackController.getEnvironmentsStack(blocker.getStackId()).getEnvironments().stream().anyMatch(environmentId -> environmentId == deployment.getEnvironmentId())) {
+                            environmentToCheck = deployment.getEnvironmentId();
+                        }
+                        break;
+                    case SERVICES:
+                        if (servicesStackController.getServicesStack(blocker.getStackId()).getServices().stream().anyMatch(serviceId -> serviceId == deployment.getServiceId())) {
+                            serviceToCheck = deployment.getServiceId();
+                        }
+                        break;
+                    default:
+                        throw new EnumConstantNotPresentException(StackType.class, "Unknown Enum type - " + stackController.getStackType(blocker.getStackId()));
```

Comment:
Why? just remove the default, you are not setting anything definitive that can cause NPE